### ISSUE TITLE
[gpu] Rocky kernel-devel is sometimes moved to the vault

### ISSUE THREAD

**Issue by cjac** [GoogleCloudDataproc/initialization-actions#1206]
When new kernels are released, the old kernel-devel (which we depend on to build kernel drivers with) is removed from the repository from which we can install.  We need to add a work-around to fetch from the vault instead.

**Comment by cjac** [comment]
```
-    execute_with_retries "dnf -y -q install pciutils kernel-devel gcc"
+    execute_with_retries "dnf -y -q install pciutils gcc"
+   if execute_with_retries "dnf -y -q install kernel-devel" ; then
+     echo "kernel-devel package installed correctly from repo"
+   elif is_rocky9 ; then
+     # this kernel-devel may have been migrated to the vault
+     vaultbase="https://download.rockylinux.org/vault/rocky/9.4/BaseOS"
+     uname_r=$(uname -r)
+     dnf --setopt=localpkg_gpgcheck=1 install \
+       ${vaultbase}/x86_64/os/Packages/k/kernel-${uname_r}.rpm \
+       ${vaultbase}/x86_64/os/Packages/k/kernel-core-${uname_r}.rpm \
+       ${vaultbase}/x86_64/os/Packages/k/kernel-modules-${uname_r}.rpm \
+       ${vaultbase}/x86_64/os/Packages/k/kernel-modules-core-${uname_r}.rpm \
+       ${vaultbase}/x86_64/os/Packages/k/kernel-devel-${uname_r}.rpm
+   fi
```

**Comment by cjac** [comment]
Thanks go to skip77 from #rockylinux on libera for the work-around here

**Comment by cjac** [comment]
see PR #1207 for review

**Comment by cjac** [comment]
Fixed with #1207
